File: backend/scratch/test_live_auth.py
import requests
import json

BASE_URL = "https://paxly-lite.onrender.com/api"

def test_auth():
    print("--- TESTING SIGNUP ---")
    signup_data = {
        "name": "Test User",
        "email": "test_auth_final@vlynxly.com",
        "password": "Password123!",
        "role": "user"
    }
    try:
        r = requests.post(f"{BASE_URL}/auth/register", json=signup_data)
        print(f"Status: {r.status_code}")
        print(f"Response: {r.text}")
        
        if r.status_code in [201, 200] or "already registered" in r.text:
            print("\n--- TESTING LOGIN ---")
            login_data = {
                "email": "test_auth_final@vlynxly.com",
                "password": "Password123!"
            }
            rl = requests.post(f"{BASE_URL}/auth/login", json=login_data)
            print(f"Status: {rl.status_code}")
            print(f"Response: {rl.text}")
            
            if rl.status_code == 200:
                print("\n✅ RESULT: Signup and Login are working perfectly on Live Backend!")
            else:
                print("\n❌ RESULT: Login failed.")
        else:
            print("\n❌ RESULT: Signup failed.")
            
    except Exception as e:
        print(f"ERROR: {e}")

if __name__ == "__main__":
    test_auth()

File: web/test_live_auth.js
const puppeteer = require('puppeteer');

(async () => {
  try {
    // 1. Sign up a new user on the live backend
    const randomStr = Math.random().toString(36).substring(7);
    const email = `testuser_${randomStr}@test.com`;
    console.log("Signing up with:", email);
    
    const signupRes = await fetch('https://paxly-premium-backend.onrender.com/api/auth/signup', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ name: 'Test User', email: email, password: 'password123' })
    });
    
    if (!signupRes.ok) {
        const text = await signupRes.text();
        console.error("Signup failed:", text);
        return;
    }
    const signupData = await signupRes.json();
    const token = signupData.access_token;
    console.log("Got token:", token);
    
    // 2. Fetch user profile
    const meRes = await fetch('https://paxly-premium-backend.onrender.com/api/auth/me', {
      headers: { 'Authorization': `Bearer ${token}` }
    });
    const user = await meRes.json();
    
    // 3. Open browser and set localStorage
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.setViewport({ width: 1280, height: 800 });
    
    page.on('console', msg => console.log('PAGE LOG:', msg.text()));
    page.on('pageerror', err => console.log('PAGE ERROR:', err.toString()));
    
    await page.goto('https://paxly-lite.vercel.app/');
    
    await page.evaluate((t, u) => {
      localStorage.setItem('token', t);
      localStorage.setItem('user', JSON.stringify(u));
    }, token, user);
    
    console.log("Navigating to /chat");
    await page.goto('https://paxly-lite.vercel.app/chat');
    await new Promise(r => setTimeout(r, 5000));
    await page.screenshot({ path: 'live_chat_screenshot2.png' });
    
    console.log("Navigating to /ai");
    await page.goto('https://paxly-lite.vercel.app/ai');
    await new Promise(r => setTimeout(r, 5000));
    
    await browser.close();
  } catch(e) {
    console.error(e);
  }
})();
